src/templates/plugin.routing.js

```diff
@@ -61,10 +61,24 @@ function localeRoute (route, locale) {
         !(strategy === STRATEGIES.NO_PREFIX) &&
         // no prefix for different domains
         !i18n.differentDomains
-
-    let path = (isPrefixed ? `/${locale}${route.path}` : route.path)
-    path = path.replace(/\/+$/, '') + (trailingSlash ? '/' : '') || '/'
-    localizedRoute.path = path
+    const thisRoute = this.router.resolve(route.path).route
+    const routeName = this.getRouteBaseName(thisRoute)
+    if (isPrefixed && !routeName) {
+      let path = `/${locale}${route.path}`
+      path = getReplacedPath(path)
+      localizedRoute.path = path
+    } else {
+      localizedRoute.path = getReplacedPath(route.path)
+      if (routeName) {
+        return this.router.resolve({
+          name: getLocaleRouteName(routeName, locale),
+          params: thisRoute.params,
+          query: thisRoute.query,
+          hash: thisRoute.hash,
+          path: localizedRoute.path
+        }).route
+      }
```

Comment:
I'd have to verify that with tests but this logic should be equivalent with no extra functions and less code duplication:

```suggestion
    let path = route.path
    if (isPrefixed && !routeName) {
      path = `/${locale}${path}`
    }
    localizedRoute.path = path.replace(/\/+$/, '') + (trailingSlash ? '/' : '') || '/'
    if (routeName) {
      localizedRoute = {
        name: getLocaleRouteName(routeName, locale),
        params: thisRoute.params,
        query: thisRoute.query,
        hash: thisRoute.hash,
        path: localizedRoute.path
      }
    }
```

Reply:
Tests passed successfully. Thanks.

Reply:
I have committed your code